discord/http.py

```diff
@@ -154,6 +156,7 @@ def handle_message_parameters(
     thread_name: str = MISSING,
     channel_payload: Dict[str, Any] = MISSING,
     applied_tags: Optional[SnowflakeList] = MISSING,
+    poll: Optional[Poll] = MISSING
```

Comment:
> i would disagree, since it makes pyright angry on `await ctx.send('Hello', poll=poll)` where `poll` is `Optional[Poll]`

Ah yes, you're right. Additionally, `abc.Messageable.send` should have `poll: Optional[Poll] = None` and not `poll: Poll = MISSING`, like it has now.

Reply:
> i would disagree, since it makes pyright angry on `await ctx.send('Hello', poll=poll)` where `poll` is `Optional[Poll]`

That would go back to the overload for `Context.send`. It's the same reason that you get an error when you do
![image](https://github.com/Rapptz/discord.py/assets/75498301/2b0c11fe-4e99-4364-b257-bf0ad9e4136b)

The revision suggestion only addressed `handle_message_parameters`, which denotes that `poll=None` cannot be given to either a `.send` or a `.edit` method.

Reply:
You're right, I was thrown off by the type hint for the send coroutines.

`handle_message_parameters` should have `Optional[Poll] = None` and all the send methods should have `poll: Optional[Poll] = None` in their function definition rather than `poll: Poll = MISSING`. In the `handle_message_parameters` block it should have `if poll is not None: ...`

```suggestion
    poll: Optional[Poll] = None
```
```python
def handle_message_parameters(poll: Optional[Poll] = None) -> ...:
    if poll is not None: ...
```